lib/package.gi

```diff
@@ -2203,9 +2227,43 @@ InstallGlobalFunction( ValidatePackageInfo, function( info )
         x -> IsString(x) and 0 < Length(x) and x[1] <> '=',
         "a nonempty string that does not start with `='" );
     TestMandat( record, "Date",
-        x -> IsString(x) and Length(x) = 10 and x{ [3,6] } = "//"
-                 and ForAll( x{ [1,2,4,5,7,8,9,10] }, IsDigitChar ),
-        "a string of the form `dd/mm/yyyy'" );
+        x -> IsString(x) and Length(x) = 10
+          and ( ( x{ [3,6] } = "//" and ForAll( x{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) )
+          or ( x{ [5,8] } = "--" and ForAll( x{ [1,2,3,4,6,7,9,10] }, IsDigitChar ) ) ) ,
+        "a string of the form `yyyy-mm-dd` or `dd/mm/yyyy`" );
+
+    #If the date is in the format `dd/mm/yyyy` a message is printed
+    if IsBound( record.Date ) and  record.Date{ [3,6] } = "//" then
+       Info( InfoPackageLoading, 2, Concatenation(record.PackageName, ": Please be advised to change the date format to `yyyy-mm-dd`"));
+    fi;
+
+    #check if the date is valid
+    CheckDateValidity := function(x)
+      local date;
+      if x{ [3,6] } = "//" then #for the old format split at '/'
+        date := List( SplitString( x, "/" ), Int);
+        #the format dd/mm/yyyy is ambigous and can be confused with mm/dd/yyyy
+        #if it is clear from the date that the format is mm/dd/yyyy print a message
+        if date[2] in [13..31] and date[1] in [1..12] then
+          TestMandat( record, "Date", 
+            x -> false, "a string of the form `yyyy-mm-dd` or `dd/mm/yyyy`" );
+          return true;
+        fi;
+      elif x{ [5,8] } = "--" then #for the yyyy-mm-dd format split at '-'
+        date := List( SplitString( x, "-" ), Int);
+        date := Permuted(date, (1,3)); #sort such that date=[dd,mm,yyyy]
+      fi;
+      if IsBound(date) and (not (date[1] in [1..31] and
+                   date[2] in [1..12] and
+                   date[3] > 1999 and # GAP 4 appeared in 1999
+                   date[1] <= DaysInMonth( date[2], date[3] ))) then
+              return false;
+      fi;
+      return true;
+    end;
+    TestMandat( record, "Date", 
+      x -> IsString(x) and CheckDateValidity(x), "a valid date");
```

Comment:
I think you should show more information in the warning:
```suggestion
      x -> IsString(x) and CheckDateValidity(x), "a string of the form `yyyy-mm-dd` or `dd/mm/yyyy` representing a date since 1999");
```